lib/evmone/execution_state.hpp

```diff
@@ -69,28 +69,33 @@ struct Stack
 
 /// The EVM memory.
 ///
-/// At this point it is a wrapper for std::vector<uint8_t> with initial allocation of 4k.
+/// The implementations uses initial allocation of 4k and the grows capacity with 2x factor.
 /// Some benchmarks has been done to confirm 4k is ok-ish value.
-/// Also std::basic_string<uint8_t> has been tried but not faster and we don't want SSO
-/// if initial_capacity is used.
-/// In future, transition to std::realloc() + std::free() planned.
 class Memory
 {
-    /// The initial memory allocation.
-    static constexpr size_t initial_capacity = 4 * 1024;
+    /// Pointer to allocated memory.
+    uint8_t* m_data = nullptr;
 
-    std::vector<uint8_t> m_memory;
+    /// The "virtual" size of the memory.
+    size_t m_size = 0;
+
+    /// The size of allocated memory. The initialization value is the initial capacity.
+    size_t m_capacity = 4 * 1024;
 
 public:
-    Memory() noexcept { m_memory.reserve(initial_capacity); }
+    /// Creates Memory object with initial capacity allocation.
+    Memory() noexcept { m_data = static_cast<uint8_t*>(std::malloc(m_capacity)); }
```

Comment:
Both here and at `realloc`, allocation failure is ignored. It is unlikely, but shouldn't there be an assert or something?

Reply:
Added something which is not horrible.